### PR TITLE
feat(wandb): log in sync with Trainer step

### LINKED ISSUE #4113: training curves look fragmented / not clear what is "step"
## ❓ train log looks fragmented / docs not clear on how to log multiple parameters

logs

The docs show how single metric is logged, what is the right way to log multiple metrics? 

Also examples show the dictionary of metrics being returned as results of `training_step` and `validation_step` functions - can't logging be done from there automatically, and if logging is done manually why do we need to return those dicts? 

#### Code
```
    def training_step(self, batch, batch_idx):
        s1, s2, target = batch
        logits = self(s1, s2)
        loss = F.cross_entropy(logits, target)
        acc = accuracy(logits, target)
        self.log('train_loss', loss)
        self.log('train_acc', acc)
        result = OrderedDict({
            'train_loss': loss,
            'train_acc': acc,
        })
        return result

    def validation_step(self, batch, batch_idx):
        s1, s2, target = batch

        logits = self(s1, s2)
        loss = F.cross_entropy(logits, target)
        acc = accuracy(logits, target)
        self.log('val_loss', loss)
        self.log('val_acc', acc)
        result = OrderedDict({
            'val_loss': loss,
            'val_acc': acc,
        })
        return result
```

#### What's your environment?

using PL 0.9.1rc4 and wandb 0.10.4

### PR DESCRIPTION
**Pull request by borisdayma**
## What does this PR do?

Uses directly `Trainer` step to log metrics and keep it in sync with wandb step.

They were decoupled due to an [issue on k-fold logging](https://github.com/PyTorchLightning/pytorch-lightning/issues/2015), where we use multiple `Trainer` for the same W&B run.

This created confusion for some users as `wandb` was then using `global_step` as a parameter to refer to `Trainer` step.
`global_step` is not used anymore and instead, we offset `step` when logging additional `Trainer` loops on the same W&B run.

Fixes #4113 

## Before submitting
- [X] Was this discussed/approved via a Github issue? (related to issue)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [X] Did you make sure to update the documentation with your changes?
- [X] Did you write any new necessary tests? 
- [X] Did you verify new and existing tests pass locally with your changes?
- [X] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [X] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: pytorch_lightning/loggers/wandb.py [hunk 1]
```diff
@@ -141,8 +143,7 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
     @rank_zero_only
     def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
         assert rank_zero_only.rank == 0, 'experiment tried to log from global_rank != 0'
-
-        self.experiment.log({'global_step': step, **metrics} if step is not None else metrics)
+        self.experiment.log(metrics, step=step + self._step_offset if step is not None else None)
```
Comment (awaelchli) [inline pytorch_lightning/loggers/wandb.py:146]:
```suggestion
        self.experiment.log(metrics, step=(step + self._step_offset) if step is not None else None)
```
for visual clarity :)

**Review by awaelchli — APPROVED** [review]
nice

**Review by rohitgr7 — APPROVED** [review]
